src/sentry/eventstream/snuba.py

```diff
@@ -13,7 +13,7 @@ def insert(self, group, event, is_new, is_sample, is_regression,
             'project_id': event.project_id,
             # TODO(mitsuhiko): We do not want to send this incorrect
             # message but this is what snuba needs at the moment.
-            'message': event.message,
+            'message': event.search_message,
```

Comment:
OK so just to be clear, after this is shipped, there will be a payload sent to snuba that looks like
```
{
...
'message': 'The big long concatenated search_message',
'data': {
    ...
    'message': 'The short display message (real_message)'
}
```
And the snuba processor should put `payload['message']` into the `search_message` column, and `payload['data']['message']` into the `message` column ?

Reply:
The changes in this file are known to not be correct. I thought the `snuba.py` was no longer used that's why I stopped updating it. I will align it with kafka.py

Reply:
OK, as an aside, I think we eventually want to deprecate most of the top level properties in the kafka payload that are duplicated in the event body (event_id, platform, etc). They were originally in there because they had been stripped out of the event body by event_manager (to save space in riak?), but IIUC that has stopped, and the event body should contain basically everything that snuba needs, with a couple exceptions for metadata like `retention_days`

But all this will require a sentry->kafka->snuba protocol change, so best left for a different PR

Reply:
So the `as_dict` method now returns `message`, `title` and `location` anyways. If we use `as_dict` instead of `event.data` here those would become available in `data`. Then we only need to carry `search_message` outside which we intentionally did not want in the event data.

(The `message` being in `as_dict` is something I want to remove because it's just an alias for `logentry.formatted or logentry.message` at this point).

Reply:
ah yeah good point, I was assuming we used as_dict. Um, I'm not sure what the best change is, moving to use as_dict instead of event.data probably needs a little bit more validation/testing for me to be comfortable with it, but ultimately its sortof where I want to move.